src/transformStateWithClones.js

```diff
@@ -6,8 +6,34 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateVers = [state];
+
+  for (let i = 0; i < actions.length; i++) {
+    let newState = { ...stateVers[i] };
+
+    switch (actions[i].type) {
+      case 'addProperties':
+        for (const [key1, value1] of Object.entries(actions[i].extraData)) {
+          newState[key1] = value1;
+        }
+        break;
+      case 'removeProperties':
+        for (const key2 of actions[i].keysToRemove) {
+          if (newState.hasOwnProperty(key2)) {
```

Comment:
I think, if you will delete undefined value it's will still work, so you can remove this check